Mailer/app/api/v1/Handlers/Mailers/Traits/ParseLogFilterConditionsTrait.php

```diff
@@ -1,33 +1,54 @@
 <?php
 
+namespace Remp\MailerModule\Api\v1\Handlers\Mailers\Traits;
 
-use Nette\Database\Table\Selection;
+use Nette\Utils\DateTime;
 
 trait ParseLogFilterConditionsTrait
 {
-    public function parseConditions(array $filter){
+    /**
+     * @param array $filter
+     * @return array
+     * @throws \Exception
+     * @example
+     * ["delivered_at", "clicked_at", "opened_at"....]
+     * or
+     * ["delivered_at" => ["from" => 2020-01-01, "to" => 2020-04-05], "clicked_at" => ["from" => 2020-01-01, "to" => 2020-04-05]]
+     */
+    public function parseConditions(array $filter)
+    {
 
         $conditions = [];
 
+        $availableColumns = ["delivered_at", "clicked_at", "opened_at", "dropped_at", "spam_complained_at", "hard_bounced_at"];
+
         // DB column can be either key in case of array input or value in case of object input
 
-        foreach ($filter as $key => $value){
-            if (is_array($value)){
-                if (isset($value['from']) && isset($value['to'])){
-                    $conditions["$key BETWEEN ? AND ?"] = [$value['from'], $value['to'] . ' 23:59:59'];
-                } else if (isset($value['from'])){
-                    $conditions["$key >="] = $value['from'];
+        foreach ($filter as $key => $value) {
+            if (is_array($value) && in_array($key, $availableColumns)) {
+                if (isset($value['from']) && isset($value['to'])) {
+                    $from = DateTime::from($value['from']);
+                    $to = DateTime::from($value['from'])->setTime(23, 59, 59);
+
+                    $conditions["$key BETWEEN ? AND ?"] = [$from, $to];
+                } else if (isset($value['from'])) {
+                    $from = DateTime::from($value['from']);
+
+                    $conditions["$key >="] = $from;
                 } else if (isset($value['to'])) {
-                    $conditions["$key <="] = $value['to'] . ' 23:59:59';
+                    $to = DateTime::from($value['from'])->setTime(23, 59, 59);
+
+                    $conditions["$key <="] = $to;
                 } else {
                     $conditions["$key NOT"] = null;
                 }
-            } else {
+            } else if (in_array($value, $availableColumns)) {
```

Comment:
Please use `elseif` instead of `else if` (I though sniffer would catch this, maybe Mailer doesn't have consistent rules with the rest of our projects)